mantle/platform/api/aws/images.go

```diff
@@ -329,10 +330,15 @@ func (a *API) CreateImportRole(bucket string) error {
 }
 
 func (a *API) CreateHVMImage(snapshotID string, diskSizeGiB uint, name string, description string) (string, error) {
+	arch := system.RpmArch()
```

Comment:
I can see why you went this path, but for maintainability, I would recommend using AWS SDK that include s `service/ec2/api.go` that has: 
```              
const ( 
    // ArchitectureTypeI386 is a ArchitectureType enum value
    ArchitectureTypeI386 = "i386"
        
    // ArchitectureTypeX8664 is a ArchitectureType enum value
    ArchitectureTypeX8664 = "x86_64"
        
    // ArchitectureTypeArm64 is a ArchitectureType enum value                                                                     
    ArchitectureTypeArm64 = "arm64"
)                           
```

Something like the following might be cleaner.
```
      var awsArch string
      switch runtime.GOARCH {
      case "amd64": 
              awsArch = api.ArchitectureTypeX8664
      case "arm64":
              awsArch = api.ArchitectureTypeArm64
      default:
              return nil, errors.News("unsupported ec2 architecture")
      }
```

The problem with the stream-data is that we end up with a confusion mapping between the Fedora/RH architecture name to the AWS architecture name.

Reply:
ah..makes sense..and much cleaner. thanks @darkmuggle